### ISSUE TITLE
`//test/extensions/filters/http/tap:tap_filter_integration_test` is flakey

### ISSUE THREAD

**Issue by phlax** [envoyproxy/envoy#25663]
This test has been flakey for a long time, albeit infrequently

It affects release branches also - seemingly more often which may indicate some caching issue

An example CI failure:

```console

AIL: //test/extensions/filters/http/tap:tap_filter_integration_test (see /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/execroot/envoy/bazel-out/aarch64-opt/testlogs/test/extensions/filters/http/tap/tap_filter_integration_test/test_attempts/attempt_1.log)

FLAKY: //test/extensions/filters/http/tap:tap_filter_integration_test (Summary)
      /build/tmp/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/execroot/envoy/bazel-out/aarch64-opt/testlogs/test/extensions/filters/http/tap/tap_filter_integration_test/test_attempts/attempt_1.log
INFO: From Testing //test/extensions/filters/http/tap:tap_filter_integration_test:
==================== Test output for //test/extensions/filters/http/tap:tap_filter_integration_test:
[==========] Running 40 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 40 tests from IpVersions/TapIntegrationTest
[ RUN      ] IpVersions/TapIntegrationTest.StaticFilePerTap/IPv4
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_no_extension_lookup_by_name to: true
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_runtime_initialized to: false
[       OK ] IpVersions/TapIntegrationTest.StaticFilePerTap/IPv4 (82 ms)
[ RUN      ] IpVersions/TapIntegrationTest.StaticFilePerTap/IPv6
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_no_extension_lookup_by_name to: true
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_runtime_initialized to: false
[       OK ] IpVersions/TapIntegrationTest.StaticFilePerTap/IPv6 (60 ms)
[ RUN      ] IpVersions/TapIntegrationTest.StaticFilePerTapWithMatchConfigAndMatch/IPv4
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_no_extension_lookup_by_name to: true
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_runtime_initialized to: false
[       OK ] IpVersions/TapIntegrationTest.StaticFilePerTapWithMatchConfigAndMatch/IPv4 (67 ms)
[ RUN      ] IpVersions/TapIntegrationTest.StaticFilePerTapWithMatchConfigAndMatch/IPv6
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_no_extension_lookup_by_name to: true
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_runtime_initialized to: false
[       OK ] IpVersions/TapIntegrationTest.StaticFilePerTapWithMatchConfigAndMatch/IPv6 (48 ms)
[ RUN      ] IpVersions/TapIntegrationTest.StaticFilePerTapWithMatchConfig/IPv4
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_no_extension_lookup_by_name to: true
[external/com_google_absl/absl/flags/internal/flag.cc : 115] RAW: Restore saved value of envoy_reloadable_features_runtime_initialized to: false
[       OK ] IpVersions/TapIntegrationTest.StaticFilePerTapWithMatchConfig/IPv4 (47 ms)
[ RUN      ] IpVersions/TapIntegrationTest.StaticFilePerTapWithMatchConfig/IPv6
[2023-02-20 09:25:53.441][16470][critical][backtrace] [./source/server/backtrace.h:96] #10: Envoy::TestRunner::RunTests() [0x151db58]
[2023-02-20 09:25:53.451][16470][critical][backtrace] [./source/server/backtrace.h:96] #11: main [0x151c900]
[2023-02-20 09:25:53.451][16470][critical][backtrace] [./source/server/backtrace.h:96] #12: __libc_start_main [0xffffa26bbe10]
================================================================================

```

https://dev.azure.com/cncf/envoy/_build/results?buildId=128658&view=logs&j=767be981-567e-57d8-68c3-2140ede0a0bd&t=1fb0b72f-6293-562d-a4d1-bb649223eaca&l=567

not sure if this is related to #13132 

i havent been able to repro locally

**Comment by phlax** [comment]
cc @daixiang0 @mattklein123

**Comment by phlax** [comment]
> ... I can not see any info in ...

seems that log has disappeared

i see it reasonably often so i will post next time i do

**Comment by phlax** [comment]
ive seen this again on the 1.22 branch but not by itself anywhere else, other than as part of failing tsan tests (#25760 )

**Comment by phlax** [comment]
here is an example on the 1.25 branch https://dev.azure.com/cncf/envoy/_build/results?buildId=129440&view=logs&j=767be981-567e-57d8-68c3-2140ede0a0bd&t=1fb0b72f-6293-562d-a4d1-bb649223eaca&l=3614

hopefully it will be retained

i have been seeing similar with local (tsan) testing - i guess due to not using RBE (#25760) - i need to figure out better why this is happening

**Comment by phlax** [comment]
i think this has been resolved by increasing the sharding #25956 - ill reopen if it seems there is a more specific issue
